### PR TITLE
[bitnami/mlflow] Adding missing Azure Storage configuration in the Helm Chart

### LINKED ISSUE #28569: [bitnami/mlflow] Add support for Azure Storage
### Name and Version

bitnami/mlflow 1.4.12

### What is the problem this feature will solve?

The MLFlow chart uses a docker image that do not support using azure storage accounts for storing artifacts, while MLFlow itself does support it.

Adding such support will make it easier to integrate MLFlow artifacts into the Microsoft cloud storage. Currently there is support for Amazon S3 and newly for Google GCS in the `2.14.1-debian-12-r1` image tag, see #22720.

### What is the feature you are proposing to solve the problem?

Adding the missing Python module `azure-storage-blob` to the mlflow image will fix the issue since MLFlow already provides support

### What alternatives have you considered?

I have built a light image using the original bitnami mlflow image as base while including the `azure-storage-blob` module

### PR DESCRIPTION
**Pull request by eriadam**
### Description of the change

The PR adds the missing configuration options for the Azure Storage Account in the MLFlow helm chart.

### Benefits

As explained in the ticket #28569, Azure Storage is supported by MLFlow, but it is not currently possible to configure it via the Helm chart. With this change Azure Blob Storage can be configured to store the artefacts of MLFlow.

### Applicable issues

- fixes #28569

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by eriadam** [comment]
@carrodher could you please have a look at this?

**Comment by carrodher** [comment]
Sorry, we didn't take a look because it is in Draft state:
```
This pull request is still a work in progress
Draft pull requests cannot be merged.
```

If you think it is ready, could you please mark it as _Ready for review_?

**Comment by eriadam** [comment]
@carrodher Thanks, it was a draft because of the last open question in here: https://github.com/bitnami/charts/issues/28569#issuecomment-2377198138

Any feedback on that is much appreciated!

**Comment by carrodher** [comment]
Thanks! To review it in the proper status, could you please rebase from the latest changes? Apart from that, please note it's needed to grant permissions to @bitnami-bot to update your fork so it can automatically update the README, CRDs, and CHANGELOG

**Comment by eriadam** [comment]
OK, I think both are done!

**Comment by eriadam** [comment]
@carrodher I did both. I added bitnami bot as a member, is that OK?

**Review by jotamartos — CHANGES_REQUESTED** [review]
Changes LGTM but you need to rebase your branch with the latest changes in stable due to there was a new version of MLflow and there are conflicts.

**Comment by eriadam** [comment]
@jotamartos Could you please have a look?

File: bitnami/mlflow/Chart.yaml [hunk 1]
```diff
@@ -44,4 +44,4 @@ sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/mlflow
 - https://github.com/bitnami/containers/tree/main/bitnami/mlflow
 - https://github.com/mlflow/mlflow
-version: 2.1.0
+version: 2.1.1
```
Comment (jotamartos) [inline bitnami/mlflow/Chart.yaml:47]:
```suggestion
version: 2.2.0
```

File: bitnami/mlflow/templates/tracking/deployment.yaml [hunk 2]
```diff
@@ -177,6 +182,7 @@ spec:
                   name: {{ include "mlflow.v0.database.secretName" . }}
                   key: {{ include "mlflow.v0.database.passwordKey" . | quote }}
             {{- end }}
+
```
Comment (jotamartos) [inline bitnami/mlflow/templates/tracking/deployment.yaml:185]:
```suggestion
```

File: bitnami/mlflow/templates/tracking/deployment.yaml [hunk 3]
```diff
@@ -191,20 +197,65 @@ spec:
                   key: {{ include "mlflow.v0.s3.secretAccessKeyKey" . | quote }}
             {{- end }}
             {{- end }}
+
```
Comment (jotamartos) [inline bitnami/mlflow/templates/tracking/deployment.yaml:200]:
```suggestion
```

File: bitnami/mlflow/templates/tracking/deployment.yaml [hunk 4]
```diff
@@ -191,20 +197,65 @@ spec:
...
             {{- if (include "mlflow.v0.s3.enabled" .) }}
             - name: MLFLOW_S3_ENDPOINT_URL
               value: {{ printf "%s://%s:%v" (include "mlflow.v0.s3.protocol" .) (include "mlflow.v0.s3.host" .) (include "mlflow.v0.s3.port" .) | quote }}
             {{- end }}
             {{- if (include "mlflow.v0.gcs.enabled" .) }}
+
```
Comment (jotamartos) [inline bitnami/mlflow/templates/tracking/deployment.yaml:206]:
```suggestion
```

**Review by jotamartos — CHANGES_REQUESTED** [review]

File: bitnami/mlflow/Chart.yaml (hunk 1 above)
Comment (jotamartos) [reply bitnami/mlflow/Chart.yaml:47]: Please bump the chart to a new minor as you are adding new parameters

**Comment by eriadam** [comment]
@ jotamartos Thank you, I added the changes.

**Review by jotamartos — APPROVED** [review]

**Pull request merged**